sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/ExternalCatalogUtils.scala

```diff
@@ -148,7 +147,7 @@ object ExternalCatalogUtils {
       }
 
       val boundPredicate =
-        InterpretedPredicate.create(predicates.reduce(And).transform {
+        Predicate.create(predicates.reduce(And).transform {
```

Comment:
Any reason to use the interpreted version here?

Reply:
Sometimes it's better to use the interpreted version if the input data is known to be small. Codegen can be slower as compiling takes time.

Reply:
Ah, ok. I'll revert this.